tools/release_tool.py

```diff
@@ -0,0 +1,355 @@
+    log(f"*** INITIALIZING RELEASE ***")
+    if args.clone:
+        clone_repo(args.repo_url, args.repo_dir)
+    else:
+        fetch_repo(args.repo_dir)
+    base_version = get_base_version(args.base_version)
+    target_version = get_target_version(base_version, args.type)
+    branch = get_branch(target_version)
+    tag = get_tag(target_version)
+    args.ref = args.ref or "main"
+    write_conf(args.type, base_version, target_version, branch, tag, args.ref)
+    if args.type == "patch":
+        switch_branch("release", branch)
+    else:
+        create_branch("release", branch, args.ref, force=args.force)
+
+
+def cherry(args):
+    log(f"*** CHERRY PICKING COMMITS ***")
+    conf = read_conf(args)
+    if args.commit:
+        run_command("Cherry picking commits", f"git cherry-pick -x {' '.join(args.commit)}")
+
+
+def changelog(args):
+    log(f"*** UPDATING CHANGELOG ***")
+    conf = read_conf(args)
+    # There are two possibilities for changelog items:
+    # * All unreleased items are going in this release (more common, when the release contains the current main branch).
+    #   In this case, we expect a single heading "Unreleased", containing all unreleased items, which will be promoted
+    #   to a versioned section; for future items, a new "Unreleased" section will be generated.
+    # * A subset of unreleased items are going in this release (less common, when the release omits some changes).
+    #   In this case, we expect a heading "Unreleased (<VERSION>)", containing the subset of unreleased items going in
+    #   this release, which will be promoted to a versioned section, and another "Unreleased" section containing the
+    #   remaining unreleased items, which will be left as is (for future items).
+    full_heading_regex = "^## \\[Unreleased\\](.*)$"
+    escaped_target_version = str(conf.target_version).replace(".", "\\.")
+    subset_heading_regex = f"^## \\[Unreleased ({escaped_target_version})\\](.*)$"
+    subset = run_command("Looking for subset heading", f"grep -q '{subset_heading_regex}' CHANGELOG.md && echo True || echo False") == "True"
+    heading_regex = subset_heading_regex if subset else full_heading_regex
+    base_tag = get_tag(conf.base_version)
+    diff_url = f"https://github.com/opendp/opendp/compare/{base_tag}...{conf.tag}"
+    replacement = f"## [{conf.target_version}] - {conf.date}\\n[{conf.target_version}]: {diff_url}"
+    # If this isn't a subset, prepend a new unreleased section.
+    if not subset:
+        replacement = "## [Unreleased](https://github.com/opendp/opendp/compare/stable...HEAD)\\n\\n\\n" + replacement
+    substitution_arg = f"-e 's|{heading_regex}|{replacement}|'"
+    inplace_arg = "-i ''" if platform.system() == "Darwin" else "-i"
+    run_command("Updating CHANGELOG", f"sed {inplace_arg} {substitution_arg} CHANGELOG.md")
+    commit("CHANGELOG", ["CHANGELOG.md"], f"RELEASE_TOOL: Updated CHANGELOG.md for {conf.target_version}.")
+
+
+def version(args):
```

Comment:
This is unused, but I'm ok with leaving it in .


Could come in handy later down the line if the libs desync and the windows build issues reappear.